tuxemon/states/pc/__init__.py

```diff
@@ -359,3 +364,59 @@ def animate_close(self) -> Animation:
         ani.update_callback = self.update_animation_position
 
         return ani
+
+
+class MonsterBoxChooseStorageState(MonsterBoxChooseState):
+    """Menu to choose a box, which you can then take a tuxemon from."""
+
+    def get_menu_items_map(self) -> Sequence[Tuple[str, MenuGameObj]]:
+        player = local_session.player
+        menu_items_map = []
+        for box_name, monsters in player.monster_boxes.items():
+            if not monsters:
+                menu_callback = partial(
+                    open_dialog,
+                    local_session,
+                    [T.translate("menu_storage_empty_kennel")],
+                )
+            else:
+                menu_callback = self.change_state(
+                    "MonsterTakeState", box_name=box_name
+                )
+            menu_items_map.append((box_name, menu_callback))
+        return menu_items_map
+
+
+class MonsterBoxChooseDropoffState(MonsterBoxChooseState):
+    """Menu to choose a box, which you can then drop off a tuxemon into."""
+
+    def get_menu_items_map(self) -> Sequence[Tuple[str, MenuGameObj]]:
+        player = local_session.player
+        menu_items_map = []
+        for box_name, monsters in player.monster_boxes.items():
+            menu_callback = self.change_state(
+                "MonsterDropOffState", box_name=box_name
+            )
+            menu_items_map.append((box_name, menu_callback))
+        return menu_items_map
+
+
+class MonsterDropOffState(MonsterMenuState):
+    """Shows all Tuxemon in player's party, puts it into box if selected."""
+
+    def startup(self, box_name: String, **kwargs: Any) -> None:
+        super().startup(**kwargs)
+
+        self.box_name = box_name
+
+    def on_menu_selection(
+        self,
+        menu_item: MenuItem[Optional[Monster]],
```

Comment:
Us monster actually optional?

Reply:
In this menu, it's definitely not Optional, but it's over-riding a method from MonsterMenuState, so I just copied the signature from there, where it has an Optional (although I'm not sure that's right). (In tuxemon/states/monster/__init__.py ) 

Should I remove the Optional for this menu?

Reply:
That will cause mypy to complain about something else. Add `assert monster` to the next line after defining monster